### ISSUE TITLE
Snowflake UNPIVOT produces parsing errors

### ISSUE THREAD

**Issue by michael-the1** [sqlfluff/sqlfluff#1474]
UNPIVOT statements are not being parsed correctly.

## Expected Behaviour

No parsing error. The UNPIVOT statement to be parsed as a `from_unpivot_expression`.

## Observed Behaviour

The UNPIVOT keyword is interpreted as an `identifier` in an `alias_expression`. The remaining part is treated as unparseable.

Example:

```sql
SELECT *
FROM tbl
UNPIVOT (f_val for f_id IN (a, b));
```

This parsing error does disappear when we add an actual alias expression, as in the next example:

```sql
SELECT *
FROM tbl AS tbl
UNPIVOT (f_val for f_id IN (a, b));
```

## Steps to Reproduce

You can run `sqlfluff lint` on the above example.
`sqlfluff parse` reveals that the UNPIVOT keyword is parsed as an `identifier` in an `alias_expression`.

## Dialect

Snowflake

## Version
python 3.9.6
sqlfluff 0.6.6

**Comment by tunetheweb** [comment]
Ah we've seen this in the past. Short term fix is to move the `UNPIVOT` from `unreserved_keywords` to `reserved_keywords` so it can't be used as a literally. Not technically correct but easiest short term fix. The problem is the parse struggles to understand which context unreserved keywords are being used in.

**Comment by tunetheweb** [comment]
BTW `PIVOT` is a `reserved_keywords` presumably for this reason.

Do you fancy doing a PR for this? Just need to:
1. Update `src/sqlfluff/dialects/dialect_snowflake.py`
2. Add a test case to `test/fixtures/parser/snowflake/snowflake_pivot.sql`
3. Run `python test/generate_parse_fixture_yml.py` to regenerate the test case YML
4. Optionally run `python test/generate_parse_fixture_yml.py` to run the test suite locally.
4. Submit the PR (which will run the test suite via GitHub actions)!

**Comment by michael-the1** [comment]
Sounds pretty straightforward! I'll put up a PR this week.
